src/internal/clusterstate/v2.8.0/pfsdb.go

```diff
@@ -212,6 +212,104 @@ func migrateRepos(ctx context.Context, tx *pachsql.Tx) error {
 	return nil
 }
 
+// alterCommitsTable adds useful new columns to pfs.commits table.
+// Note that this is not the end all be all. We will need to make more changes after data has been migrated.
+// TODO
+//   - rename int_id to id? This will requires changing all references as well.
+//   - make repo_id not null
+//   - make origin not null
+//   - make updated_at not null and default to current timestamp
+//   - branch_id_str is a metadata reference to the branches table. Today this points to collections, but tomorrow it should
+//     point to pfs.branches. Once the PFS master watches branches, we will no longer need this column.
+func alterCommitsTable(ctx context.Context, tx *pachsql.Tx) error {
+	query := `
+	CREATE TYPE pfs.commit_origin AS ENUM ('ORIGIN_KIND_UNKNOWN', 'USER', 'AUTO', 'FSCK');
+
+	DROP INDEX IF EXISTS pfs.commit_provenance_from_id_idx;
+
+	ALTER TABLE pfs.commits
+		ADD COLUMN repo_id bigint REFERENCES pfs.repos(id),
```

Comment:
```suggestion
		ADD COLUMN repo_id bigint NOT NULL REFERENCES pfs.repos(id),
```

Reply:
I can't do this change now because the field will be empty until after the data migration.